src/Symfony/Component/HttpKernel/DependencyInjection/ResettableServicePass.php

```diff
@@ -57,6 +57,10 @@ public function process(ContainerBuilder $container)
                     $methods[$id] = [];
                 }
 
+                if ('ignore' === ($attributes['on_invalid'] ?? null)) {
+                    $attributes['method'] = '?'.$attributes['method'];
```

Comment:
Then, why not allowing 
```php
->tag('kernel.reset', ['method' => '?reset']
``` 
directly?

Reply:
less explicit?

Reply:
Actually, it should already be working with the current code